src/main/java/com/google/api/codegen/gapic/GapicProviderFactory.java

```diff
@@ -0,0 +1,202 @@
+          .build();
+
+      return Arrays.<GapicProvider<? extends Object>>asList(mainProvider, packageInfoProvider);
+
+    } else if (id.equals("nodejs")) {
+      GapicProvider<? extends Object> mainProvider = CommonGapicProvider.<Interface>newBuilder()
+          .setModel(model)
+          .setView(new InterfaceView())
+          .setContext(new NodeJSGapicContext(model, apiConfig))
+          .setSnippetSetRunner(new NodeJSSnippetSetRunner<Interface>())
+          .setSnippetFileNames(Arrays.asList("main.snip"))
+          .build();
+      GapicProvider<? extends Object> clientConfigProvider = CommonGapicProvider.<Interface>newBuilder()
+          .setModel(model)
+          .setView(new InterfaceView())
+          .setContext(new ClientConfigGapicContext(model, apiConfig))
+          .setSnippetSetRunner(new ClientConfigSnippetSetRunner<Interface>())
+          .setSnippetFileNames(Arrays.asList("json.snip"))
+          .build();
+
+      return Arrays.<GapicProvider<? extends Object>>asList(mainProvider, clientConfigProvider);
+
+    } else if (id.equals("php")) {
+      GapicProvider<? extends Object> provider = CommonGapicProvider.<Interface>newBuilder()
+          .setModel(model)
+          .setView(new InterfaceView())
+          .setContext(new PhpGapicContext(model, apiConfig))
+          .setSnippetSetRunner(new PhpSnippetSetRunner<Interface>())
+          .setSnippetFileNames(Arrays.asList("main.snip"))
+          .build();
+      return Arrays.<GapicProvider<? extends Object>>asList(provider);
+
+    } else if (id.equals("python")) {
+      GapicProvider<? extends Object> mainProvider = CommonGapicProvider.<Interface>newBuilder()
+          .setModel(model)
+          .setView(new InterfaceView())
+          .setContext(new PythonGapicContext(model, apiConfig))
+          .setSnippetSetRunner(new PythonSnippetSetRunner<Interface>(new PythonInterfaceInitializer()))
+          .setSnippetFileNames(Arrays.asList("main.snip"))
+          .build();
+      GapicProvider<? extends Object> messageProvider = CommonGapicProvider.<ProtoFile>newBuilder()
+          .setModel(model)
+          .setView(new ProtoFileView())
+          .setContext(new PythonGapicContext(model, apiConfig))
+          .setSnippetSetRunner(new PythonSnippetSetRunner<ProtoFile>(new PythonProtoFileInitializer()))
+          .setSnippetFileNames(Arrays.asList("message.snip"))
+          .setOutputSubPath("")
+          .build();
+      GapicProvider<? extends Object> clientConfigProvider = CommonGapicProvider.<Interface>newBuilder()
+          .setModel(model)
+          .setView(new InterfaceView())
+          .setContext(new ClientConfigGapicContext(model, apiConfig))
+          .setSnippetSetRunner(new ClientConfigSnippetSetRunner<Interface>())
+          .setSnippetFileNames(Arrays.asList("json.snip"))
+          .build();
+
+      return Arrays.<GapicProvider<? extends Object>> asList(mainProvider, messageProvider,
```

Comment:
nit: no space after the parameterized type, here and below. (The Java compiler can't infer the here from the parameters?)